go/registry/api/api.go

```diff
@@ -475,26 +475,32 @@ func VerifyRegisterNodeArgs( // nolint: gocyclo
 	}
 	expectedSigners = append(expectedSigners, n.ID)
 	if !inEntityNodeList {
-		// Entity signing node registrations is feature-gated by a consensus
-		// parameter, and a per-entity configuration option.
-		if !params.DebugAllowEntitySignedNodeRegistration || !entity.AllowEntitySignedNodes {
-			logger.Error("RegisterNode: registration likely signed by entity",
-				"signed_node", sigNode,
-				"node", n,
-			)
-			return nil, nil, fmt.Errorf("%w: registration likely signed by entity", ErrInvalidArgument)
-		}
-
 		// If we are using entity signing, descriptors will also be signed
 		// by the entity signing key.
-		if !sigNode.MultiSigned.IsSignedBy(entity.ID) {
+		if sigNode.MultiSigned.IsSignedBy(entity.ID) {
+			// Entity signing node registrations is feature-gated by a consensus
+			// parameter, and a per-entity configuration option.
+			//
+			// Ignore the per-entity option during sanity checks as the entity
+			// could have been modified (unless this is during genesis).
+			if !params.DebugAllowEntitySignedNodeRegistration || (!entity.AllowEntitySignedNodes && (!isSanityCheck || isGenesis)) {
+				logger.Error("RegisterNode: registration signed by entity",
+					"signed_node", sigNode,
+					"node", n,
+				)
+				return nil, nil, fmt.Errorf("%w: registration signed by entity", ErrInvalidArgument)
```

Comment:
To summarize the logic leading to this:

- The node is absent from its entity's node list
- The node registration is signed by its entity

Now, there are different configurations and codepaths where we want to do different things for a node in this circumstance, but they all lead to this single `VerifyRegisterNodeArgs` function, so there's further logic.

1. There's a mode of registration where an entity signs its node's registration rather than putting the node's ID in its nodes list (:information_source: this requires the entity keys to be online, which is riskier, so we don't recommend it). In fact, this is the normal circumstance when using this mode. To use this, the network enables it with a flag, and the entity enables it with a flag. In that configuration, the registration is valid.
2. But if the network and entity are not thus configured, then we have to follow the rules for the default mode, where the node must be in its entity's node list, which it's not, and it's not valid. Depending on the codepath, we want to do different things.
   1. If we asked about validity because someone is trying to register this node right now, then turn them away, because it would be useless to add an invalid node.
   2. If we're asking about some node already in our records (a 'sanity check'), then it's not valid, but it can happen because an entity might have kicked one of its nodes out, and there's no special process to clean up the node registration. We just wait for it to expire normally, while understanding that it's not valid.
   3. However, if we're asking about an entry in a genesis file, then be extra strict and say that the node shouldn't be there, so that people don't accidentally have useless records in their genesis file.

**This patch adjusts `VerifyRegisterNodeArgs` not to return an error in situation 2.ii.**

It feels awkward from an API perspective, for a function named in this way to conclude that the registration is valid. The design that leads to all this circumstantial logic being piled into a 'verify args' routine is bursting at the seams. But on the other hand, this is a timely fix for a tool that we would like to continue running.

On a side note, it looks like the error messages returned around here are meant to be on the humanly-usable side and to guide users toward correctly setting up entity-signed nodes, but I think they could be more verbose.